x-pack/plugins/enterprise_search/public/applications/app_search/types.ts

```diff
@@ -6,3 +6,21 @@
 
 export * from '../../../common/types/app_search';
 export { IRole, TRole, TAbility } from './utils/role';
+
+export interface IApiToken {
+  access_all_engines?: boolean;
+  key?: string;
+  engines?: string[];
+  id?: number;
+  name: string;
+  read?: boolean;
+  type: string;
+  write?: boolean;
+}
```

Comment:
Super sorry, I definitely should have mentioned this yesterday as well, just slipped my mind - I did a quick search on our ent-search repo and it looks like credentials is the only component/view that uses `IApiToken`. I think that makes it a pretty good candidate to move into `credentials/types.ts` - what do you think?

Reply:
Yes, that makes a ton of sense. If we ever need it elsewhere we could pull it out.